### PR TITLE
Removed OrderedDict from io.registry

### LINKED ISSUE #10888: Remove OrderedDict for astropy 4.2
### Description

Based on https://github.com/astropy/astropy-APEs/pull/63, it looks like astropy 4.2 will support only Python 3.7+, so we should be able to count on the built-in `dict` being ordered. 

So can we get rid of all the `OrderedDict` now?

### Additional context

https://mail.python.org/pipermail/python-dev/2017-December/151283.html

### PR DESCRIPTION
**Pull request by marcellonascif**
### Description

This pull request removes OrderedDict from files in io.registry. The files are base.py and core.py.
The OrderedDicts are normal dictionaries now.

Fixes part of #10888

### CODE REVIEW

**Comment by pllim** [comment]
The "dev" failure is unrelated, so no need to worry. 😸

**Review by nstarman — APPROVED** [review]
Thanks! These states are never serialized to yaml, so there's no worry about the ordering being lost.

File: docs/changes/io.registry/15494.api.rst
[the remark this answers is not in the document]
Comment (nstarman) [reply docs/changes/io.registry/15494.api.rst:1]: No.

**Comment by pllim** [comment]
Thank you, all!

**Comment by marcellonascif** [comment]
Thank you guys again! You are helping me a lot!

**Pull request merged**